privacyidea/lib/eventhandler/usernotification.py

```diff
@@ -345,7 +351,20 @@ def do(self, action, options=None):
                 mimetype = handler_options.get("mimetype", "plain")
                 useremail = recipient.get("email")
                 reply_to = handler_options.get("reply_to")
-
+                attachment = handler_options.get("attachment", False)
+
+                if attachment and googleurl_img:
```

Comment:
Just a logical question: The "attachment" option will always attach a googleurl_img, if it exists in the response, no matter if the "googleurl_img" is contained in the body.
So the "attachment" implicitly means "attach QR Code from enrolled tokens"?
OK, the description indicates it. But maybe we might have some other "attachment" in the future and then actually the name should be more specific like "qrcode_attachment"? What do you think?

Reply:
Of course, this specifically targets the `googleurl`. There is also the `oathurl` and `otpkey` as QR-Code images in the response but i  haven't seen them used anywhere else.
So ideally we would allow to select all (QR-Code) images as attachments (and maybe some other stuff as well? PDF with descriptions, ...) but i think handling this here is overkill.
We could add attachments based on [`CID` URLs](https://tools.ietf.org/html/rfc2392) in the body. But this means we would have to parse the body before.
Maybe we can externalize the whole email creation stuff and we only provide the QR-Code images.

Reply:
I think part of my point was that "attachent" sounds like a very generic keyword. If we ever^(TM) would want to attach other things to a notification, this generic word would be wasted. 

Image we would one day send notifactions to students including a PDFs with certain information and we would attach the PDF, or we would send a notification "you are ready to enroll your smartcard, please use the attached enrollment tool" and we would attach a ZIP or EXE or whatever.
At the moment this might be far fetch but it is a feeling in my guts, with the experience with privacyIDEA, that "attachment" might be something bigger int he future.

So please use a keyword like "attach_qrcode" to be able to be more generic in the future.

Reply:
will do

Reply:
Done in 4dc32558